src/streamlink/plugins/vlive.py

```diff
@@ -1,58 +1,85 @@
+from streamlink.plugin.api import validate
 from streamlink.stream import HLSStream
+from streamlink.utils import parse_json
 
 log = logging.getLogger(__name__)
 
 
 class Vlive(Plugin):
-    _url_re = re.compile(r"https?://(?:www.)vlive\.tv/video/(\d+)")
-    _video_status = re.compile(r'oVideoStatus = (.+)<', re.DOTALL)
-    _video_init_url = "https://www.vlive.tv/video/init/view"
+    _url_re = re.compile(r"https?://(?:www\.)vlive\.tv/(?P<format>video|post)/(?P<id>[0-9\-]+)")
+    _page_info = re.compile(r'window\.__PRELOADED_STATE__\s*=\s*({.*})\s*<', re.DOTALL)
+    _playinfo_url = "https://www.vlive.tv/globalv-web/vam-web/old/v3/live/{0}/playInfo"
+
+    _schema_video = validate.Schema(
+        validate.transform(_page_info.search),
+        validate.any(None, validate.all(
+            validate.get(1),
+            validate.transform(parse_json),
+            {"postDetail": {"post": {"officialVideo": {
+                "type": str,
+                "videoSeq": int,
+                validate.optional("status"): str,
+            }}}},
+            validate.get("postDetail"),
+            validate.get("post"),
+            validate.get("officialVideo")
+        ))
+    )
+
+    _schema_stream = validate.Schema(
+        validate.transform(parse_json),
+        validate.all(
+            {"result": {"streamList": [{
+                "streamName": str,
+                "serviceUrl": str,
+            }]}},
+            validate.get("result"),
+            validate.get("streamList")
+        )
+    )
 
     @classmethod
     def can_handle_url(cls, url):
         return cls._url_re.match(url) is not None
 
-    @property
-    def video_id(self):
-        return self._url_re.match(self.url).group(1)
-
     def _get_streams(self):
-        vinit_req = self.session.http.get(self._video_init_url,
-                                          params=dict(videoSeq=self.video_id),
-                                          headers=dict(referer=self.url))
-        if vinit_req.status_code != 200:
-            raise PluginError('Could not get video init page (HTTP Status {})'
-                              .format(vinit_req.status_code))
-
-        video_status_js = self._video_status.search(vinit_req.text)
-        if not video_status_js:
-            raise PluginError('Could not find video status information!')
-
-        video_status = json.loads(video_status_js.group(1))
-
-        if video_status['viewType'] == 'vod':
-            raise PluginError('VODs are not supported')
+        video_json = self.session.http.get(self.url, headers={"Referer": self.url},
+                                           schema=self._schema_video)
+        if video_json is None:
+            log.error('Could not parse video page')
+            return
+
+        if video_json['type'] == 'VOD':
+            log.error('VODs are not supported')
+            return
+
+        url_format, video_id = re.match(self._url_re, self.url).group('format', 'id')
```

Comment:
```suggestion
        url_format, video_id = self._url_re.match(self.url).groups()
```

you don't need `re.`